repose-aggregator/commons/utilities/src/test/scala/org/openrepose/commons/utils/servlet/http/HttpServletRequestWrapperTest.scala

```diff
@@ -521,6 +523,7 @@ class HttpServletRequestWrapperTest extends FunSpec with BeforeAndAfter with Mat
       wrappedRequest.getPreferredHeader("ornament") shouldBe "santa"
     }
 
+    //todo: fix this test -- shouldn't the higher quality value be returned?
     it("should return the appropriate value when a higher quality is appended onto a line") {
```

Comment:
I'm kind of leaning toward parsing from the end of the value. Header value parameters always following the value itself, so if a value is not splittable, it kind of makes sense to return everything before the last ;q=<blargh>. This test expects everything before the first ;q=<blargh>.

To clarify, given the header value:
"Foo: bar;q=0.9,baz;q=0.8"

I would expect the return value to be:
"bar;q=0.9,baz"

Is that reasonable?